File: RemixedDungeon/src/main/java/com/watabou/pixeldungeon/actors/buffs/Stun.java

package com.watabou.pixeldungeon.actors.buffs;

import com.nyrds.pixeldungeon.ml.R;
import com.nyrds.platform.util.StringsManager;
import com.watabou.pixeldungeon.actors.Char;
import com.watabou.pixeldungeon.items.rings.RingOfElements.Resistance;
import com.watabou.pixeldungeon.sprites.CharSprite;
import com.watabou.pixeldungeon.ui.BuffIndicator;

import org.jetbrains.annotations.NotNull;

public class Stun extends FlavourBuff {

	private static final float DURATION	= 10f;
	
	@Override
	public boolean attachTo(@NotNull Char target ) {
		if (super.attachTo( target )) {
			target.paralyse(true);
			return true;
		} else {
			return false;
		}
	}
	
	@Override
	public void detach() {
		target.paralyse(false);
		super.detach();
	}
	
	@Override
	public int icon() {
		return BuffIndicator.PARALYSIS;
	}

	public static float duration( Char ch ) {
		Resistance r = ch.buff( Resistance.class );
		return r != null ? r.durationFactor() * DURATION : DURATION;
	}

	@Override
	public CharSprite.State charSpriteStatus() {
		return CharSprite.State.PARALYSED;
	}

	@Override
	public void attachVisual() {
		target.getSprite().add(CharSprite.State.PARALYSED);
        target.showStatus(CharSprite.NEGATIVE, StringsManager.getVar(R.string.Char_StaStunned));
	}
}

File: scripts/buffs/Stun.lua
---
--- Stun: paralyses while active (was actors/buffs/Stun.java)
---
local RPD  = require "scripts/lib/commonClasses"
local buff = require "scripts/lib/buff"

local CharSprite = luajava.bindClass("com.watabou.pixeldungeon.sprites.CharSprite")

return buff.init{
    desc  = function ()
        return {
            icon          = 4, -- BuffIndicator.PARALYSIS
            name          = "StunBuff_Name",
            info          = "StunBuff_Info",
        }
    end,

    attached = function(self, buff)
        buff.target:paralyse(true)
    end,

    detach = function(self, buff)
        buff.target:paralyse(false)
    end,

    charSpriteStatus = function(self, buff)
        return "PARALYSED"
    end,

    attachVisual = function(self, buff)
        buff.target:showStatus(CharSprite.NEGATIVE, RPD.textById("Char_StaStunned"))
    end
}
